index.js

```diff
@@ -311,6 +307,7 @@ function loader (registryOrVersion) {
     set durabilityUsed (value) {
       const where = registry.supportFeature('whereDurabilityIsSerialized')
       if (where === 'Damage') {
+        if (!this?.nbt?.value?.Damage && !value) return
```

Comment:
Why add this back?

Reply:
Because that's what actually fixes the issue we are facing

Reply:
This goes back to my initial comment. If you need to deviate from expected server side behavior (like setting durability should store the durability inside the items instead of being undefined) to fix an issue in the client than the fix is no good. So what happens if the durability field does exist, and you want to set the durability to max? It would have to be deleted. Issues specific to the client like that should be fixed when serializing data just to the server - like inside toNotch(). 

Separately I agree with the point that data shouldn’t be modified beyond what the server sends. The constructor tried to handle both cases of server side instantiation and also from user code, which is what caused the problem with it trying to override server data. That can be fixed like what you did here (remove changes) or adding a new static method like fromNetwork() to do the opposite of toNotch(). But since the constructor here is pretty small I agree your change just deleting this line works.

Reply:
I think it makes a lot more sense to just let the implementing library set the Damage nbt data in the constructor as it has always been until v1.13.0. Maybe we could use the feature flag in the setter (i.e. if damage is not set, value is 0 and version does not support explicitMaxDurability, we return. Else we carry on) but I don't really see the point of the setter in general. If we need to update raw NBT data at a higher level then we'd probably like to do so explicitly (to avoid issues like what we are experiencing in Mineflayer now) and the use cases for this abstraction are so few it really doesn't matter to have it anyway (as server sends the item update when it loses durability anyway). 

As mentioned earlier I am on my phone so I won't get into detail with your other points as I feel I have already made my points clear with the above and in previous comments

Reply:
The setter would be used in a flying squid plugin, or another server side application. 

The purpose of this abstraction is that damage may not be stored in the same place, so any changes between versions (like you can see in the current implementation) can be handled in this library as opposed to other user code. Maybe it’s not even inside NBT and a separate field - that’s handled. 

I don’t object to a “noDamageMeansNoNbtTag”  feature, if we were deleted the applicable NBT for that version when the durability is 0, instead of setting 0. 

> As mentioned earlier I am on my phone so I won't get into detail with your other points as I feel I have already made my points clear with the above and in previous comments

same ;)